src/builtins/utils.zig

```diff
@@ -257,25 +319,27 @@ pub fn decrefRcPtrC(
     bytes_or_null: ?[*]isize,
     alignment: u32,
     elements_refcounted: bool,
+    roc_dealloc: RocDealloc,
```

Comment:
We do not want to pass functions individual.

Fundamentally, all roc functions (that can allocate or run effects and by extension these builtins) should be passed pointer to a struct of functions pointers.

For builtins, it would be viewed as:
```zig
const RocEffects = struct {
    context: anyopaque,
    roc_alloc: RocAlloc,
    roc_realloc: RocRealloc,
    roc_dealloc: RocDealloc,
    ...
};
```
Then would be used here as `roc_effects: *const RocEffects`.

In the interpreter, we will just keep a reference to RocEffects in the call context. When compiling for llvm, we will have to make it a function argument to essentially all functions.

Reply:
To add more color, after all the required builtin functions will be pointers to all of the effect functions in alphabetic order. The platform will pass it into the app when calling an application function. The application will use it to run the various effects